pallets/ah-migrator/src/call.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::*;
+use frame_support::traits::Bounded;
+
+pub type BoundedCallOf<T> =
+	Bounded<<T as frame_system::Config>::RuntimeCall, <T as frame_system::Config>::Hashing>;
+
+impl<T: Config> Pallet<T> {
+	pub fn map_rc_ah_call(
+		rc_bounded_call: &BoundedCallOf<T>,
+	) -> Result<BoundedCallOf<T>, Error<T>> {
+		let encoded_call = if let Ok(e) = Self::fetch_preimage(rc_bounded_call) {
+			e
+		} else {
+			return Err(Error::<T>::PreimageNotFound);
+		};
+
+		if let Some(hash) = rc_bounded_call.lookup_hash() {
+			if T::Preimage::is_requested(&hash) {
+				T::Preimage::unrequest(&hash);
+			}
+		}
+
+		let call = if let Ok(call) = T::RcToAhCall::try_convert(&encoded_call) {
+			call
+		} else {
+			return Err(Error::<T>::FailedToConvertCall);
+		};
+
+		log::info!("MAPPED CALL: {:?}", call);
+
+		let ah_bounded_call = T::Preimage::bound(call).map_err(|err| {
```

Comment:
This is not noting the preimage, right? So it just returns a lookup hash?  
so when or where will we note it?

Reply:
This does `note` if it does not fit into `inline` size limit.

Reply:
I dont find where it notes it, i checked this [bound](https://github.com/paritytech/polkadot-sdk/blob/f4f56f6cf819472fcbab7ef367ec521f26cb85cb/substrate/frame/support/src/traits/preimages.rs#L247-L260) function.